notebook/static/notebook/js/notebook.js

```diff
@@ -956,11 +968,15 @@ define(function (require) {
      */
     Notebook.prototype.delete_cells = function(indices) {
         if (indices === undefined) {
-            indices = this.get_marked_indices();
-            
-            if (indices.length === 0) {
-                indices = [this.get_selected_index()];
-            }
+            indices = [this.get_selected_index()];
+        }
+        this._delete_cells(indices);
+    }
+
+
+    Notebook.prototype._delete_cells = function(indices) {
+        if(indices === undefined){
+            throw new Error('need indices to delete cells');
```

Comment:
The private one should always raise if no indices are given. The public one can do heuristics, that reduce the number of potential bugs.